test/appsignal/tracer_test.exs

```diff
@@ -482,6 +482,43 @@ defmodule Appsignal.TracerTest do
     end
   end
 
+  describe "on_create_span/2" do
+    setup do
+      on_exit(fn ->
+        Application.put_env(:appsignal, :custom_on_create_fun, &Tracer.custom_on_create_fun/1)
+      end)
+    end
+
+    test "custom data is set with custom_on_create_fun_span" do
+      Application.put_env(
+        :appsignal,
+        :custom_on_create_fun,
+        &__MODULE__.custom_on_create_fun_span/1
+      )
+
+      assert %{"sample_data" => %{"custom_data" => "{\"foo\":\"bar\"}"}} =
+               "http_request"
+               |> Tracer.create_span()
+               |> Span.to_map()
```

Comment:
This assertion is failing on CI. Locally it passes for me though.

Reply:
Same for me, could it be that somehow locally without a NIF is compiled different than on semaphore?

Reply:
Yes, I think it is correct for this test to fail in no NIF mode. `Span.to_map()` is provided via a NIF call.

Reply:
Right, then you can add the `@tag :skip_env_test_no_nif` decorator above the test definition `test "..." do` to skip it for that test environment.